pyomo/contrib/iis/iis.py

```diff
@@ -0,0 +1,136 @@
+"""
+This module contains functions for computing an irreducible infeasible set
+for a Pyomo MILP or LP using a specified commerical solver, one of CPLEX,
+Gurobi, or Xpress.
+"""
+
+import abc
+from pyomo.environ import SolverFactory
+
+
+def write_iis(pyomo_model, iis_file_name, solver=None):
+    """
+    Write an irreducible infeasible set for a Pyomo MILP or LP
+    using the specified commerical solver.
+
+    Arguments
+    ---------
+        pyomo_model:
+            A Pyomo Block or ConcreteModel
+        iis_file_name:str
+            A file name to write the IIS to, e.g., infeasible_model.ilp
+        solver:str
+            Specify the solver to use, one of "cplex", "gurobi", or "xpress".
+            If None, the tool will use the first solver available.
+
+    Returns
+    -------
+        iis_file_name:str
+            The file containing the IIS.
+    """
+
+    available_solvers = [
+        s
+        for s, sp in zip(_supported_solvers, _supported_solvers_persistent)
+        if SolverFactory(sp).available(exception_flag=False)
+    ]
+
+    if solver is None:
+        if len(available_solvers) == 0:
+            raise RuntimeError("Could not find a solver to use")
+        solver = _available_solvers[0]
+        print(f"Using solver {solver}")
```

Comment:
I would replace all print statements with logging statements so they can be turned off if needed.